### PR TITLE
Don't use deprecated request.REQUEST

### PR DESCRIPTION
**Pull request by atodorov**
This PR removes deprecated `request.REQUEST` to allow for easier upgrade of Django in the future. 

NOTES:

* There are probably other places where this is used but I haven't touched them yet.
* All tests report PASS but looking at coverage I think all of the lines which this PR changes are not covered by tests. I will see if I can write some tests but maybe not immediately.

### CODE REVIEW

**Comment by tkdchen** [comment]
Please sign-off your commit, and the commit message should tell more information about where ``request.REQUEST``s are fixed, because this patch only modifies part of the code, not all.

There are lots of ``request.REQUEST`` throughout the code. I prefer to change those code along with unit tests.

**Comment by atodorov** [comment]
> There are lots of request.REQUEST throughout the code. I prefer to change those code along with unit tests.

I totally support that. As explained in the updated commit log I have created these some time ago before the test suite was working. Then I realized how many of these changed are necessary and stopped but decided to send this PR so my work doesn't get lost.

**Comment by tkdchen** [comment]
Relative to #110

**Comment by tkdchen** [comment]
So far, I found two issues,

- Cannot add tag to selected cases in Cases tab
- Cannot add a case to plan

File: tcms/testcases/views.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def plan_from_request_or_none(request, pk_enough=False):
     Arguments:
     - pk_enough: a choice for invoker to determine whether the ID is enough.
     '''
-    tp_id = request.REQUEST.get("from_plan")
+    tp_id = request.POST.get("from_plan")
```
Comment (tkdchen) [inline tcms/testcases/views.py]: ``plan_from_request_or_none`` is used in either POST and GET request.

**Comment by atodorov** [comment]
@tkdchen if you can merge #171 so we can have good tests on various databases I will go back to this one and rework it to add more tests before making changes b/c it is obvious I have broken something.

**Comment by tkdchen** [comment]
The testing matrix in #171 does not block anything currently IMO. Please fix above two issues found in this PR so that I can merge this PR.

**Comment by atodorov** [comment]
> So far, I found two issues,
>    Cannot add tag to selected cases in Cases tab

Fixed. Removal of tag is issue #174 which is fixed in #177 

>    Cannot add a case to plan

I can't reproduce. I tried:

1) Open TestPlan, cases tab
2) Add case from other plans
3) search by case id & add ==> works

then open test case -> Plans tab, add plan by ID - also works. Can you test and tell me how to reproduce?

File: tcms/testcases/views.py [hunk 2]
```diff
@@ -468,7 +468,7 @@ def get_selected_testcases(request):
     Arguments:
     - request: REQUEST object.
     '''
-    REQ = request.REQUEST
+    REQ = request.POST
```
Comment (tkdchen) [inline tcms/testcases/views.py]:
```
REQ = request.POST or request.GET
```

This fixes ``Cannot add tag to selected cases in Cases tab``.

**Comment by tkdchen** [comment]
f956f83 includes too many other changes that are not relative to the issue found during review. And, only one line code change fixes the first issue, please see my comment above. Just update bf5041a is enough, no need of f956f83.

I tried to reproduce issue ``Cannot add a case to plan``, but also cannot reproduce it again in both "Write new case" and "Add cases from other plans".

**Comment by atodorov** [comment]
Updated. I will open another PR for the fixes from f956f83.

**Comment by tkdchen** [comment]
Merged. Thanks.

**Comment by atodorov** [comment]
Merged manually, closing. 

@tkdchen FYI you can use the merge button to Rebase & merge which will place the PR commits on top of the develop branch.

**Pull request closed without merging**